modules/gradio_ui.py

```diff
@@ -2,61 +2,73 @@
-    
-    data = {
-        "query": prompt
-    }
-
-    if not use_history:
-        print("Ignoring conversation history")
-    elif use_history and conversation_id!=None:
-        data["conversation_id"]=conversation_id
-        print("Using conversation ID: "+conversation_id)
-
-    # Convert the data dictionary to a JSON string
-    json_data = json.dumps(data)
-
-    try:
-        # Make the HTTP POST request
-        response = requests.post(url, headers=headers, data=json_data)
-
-        # Check if the request was successful (status code 200)
-        if response.status_code == 200:
-            print("Response JSON:", response.json())
-            conversation_id=response.json().get("conversation_id")
-            return response.json().get("response")
-        else:
-            print(f"Request failed with status code {response.status_code}")
-            print("Response text:", response.text)
-            return "Sorry, an error occurred: "+ response.text
-    except requests.RequestException as e:
-        # Handle any exceptions that may occur during the request
-        return f"An error occurred: {e}"
-
-
-use_history=gr.Checkbox(value=True, label="Use history")
-ui = gr.ChatInterface(yaml_gen,
-                        additional_inputs=[
-                            use_history
-                        ])
+        print("Using history: "+str(use_history))
+        # Body of the request (a JSON object with a "query" field)
+        
+        data = {
+            "query": prompt
+        }
+
+        if not use_history:
+            print("Ignoring conversation history")
+        elif use_history and self.conversation_id!=None:
+            data["conversation_id"]=self.conversation_id
+            print(f"Using conversation ID: {self.conversation_id}")
+
+        # Convert the data dictionary to a JSON string
+        json_data = json.dumps(data)
+
+        try:
+            # Make the HTTP POST request
+            response = requests.post(url, headers=headers, data=json_data)
+
+            # Check if the request was successful (status code 200)
+            if response.status_code == 200:
+                print("Response JSON:", response.json())
+                self.conversation_id=response.json().get("conversation_id")
+                return response.json().get("response")
+            else:
+                print(f"Request failed with status code {response.status_code}")
+                print(f"Response text: {response.text}")
+                return "Sorry, an error occurred: "+ response.text
+            
+        except requests.RequestException as e:
+            # Handle any exceptions that may occur during the request
+            return f"An error occurred: {e}"
 
+    def mount_ui(self, fast_api_instance):
+        return gr.mount_gradio_app(fast_api_instance, 
+                                   self.ui, 
+                                   path=self.mount_path)
 
 if __name__ == "__main__":
```

Comment:
I think we can remove this if block as it is just for testing purposes.

Reply:
For dev & test purposes, having a "main" block to test the functionality as stand-alone it is easier for the developer. It is ignored during execution when imported from another file.